FirebaseInAppMessaging/Tests/Unit/FIRIAMClientInfoFetcherTests.m

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2020 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <OCMock/OCMock.h>
+#import <XCTest/XCTest.h>
+
+#import "FIRIAMClientInfoFetcher.h"
+
+#import <FirebaseInstallations/FIRInstallations.h>
+#import <FirebaseInstallations/FIRInstallationsAuthTokenResult.h>
+
+@interface FIRInstallationsAuthTokenResult (Tests)
+- (instancetype)initWithToken:(NSString *)token expirationDate:(NSDate *)expirationDate;
+@end
+
+@interface FIRIAMClientInfoFetcherTests : XCTestCase
+
+@property(nonatomic, strong) FIRIAMClientInfoFetcher *clientInfoFetcher;
+
+@end
+
+@implementation FIRIAMClientInfoFetcherTests
+
+- (NSDate *)future {
+  return [[NSDate date] dateByAddingTimeInterval:10000];
+}
+
+- (void)setUp {
+  // Set up for the happy bath where both FID and FIS token are returned.
+  FIRInstallationsAuthTokenResult *tokenResult =
+      [[FIRInstallationsAuthTokenResult alloc] initWithToken:@"mock_token"
+                                              expirationDate:[self future]];
+  id mockInstallations = [self mockInstanceIDMethodForTokenAndIdentity:tokenResult
+                                                            tokenError:nil
+                                                              identity:@"mock_id"
+                                                         identityError:nil];
+
+  self.clientInfoFetcher =
+      [[FIRIAMClientInfoFetcher alloc] initWithFirebaseInstallations:mockInstallations];
```

Comment:
Nit: if we store `mockInstallations`  to a property we will be able to create `self.clientInfoFetcher` once and stub the mock as we need in each test (I guess the default stub in setUp method may not be needed then).